tsdb/agent/db_test.go

```diff
@@ -934,6 +934,46 @@ func TestDBOutOfOrderTimeWindow(t *testing.T) {
 	}
 }
 
+func TestDBCreatedTimestampSamplesIngestion(t *testing.T) {
+	reg := prometheus.NewRegistry()
+	opts := DefaultOptions()
+	opts.OutOfOrderTimeWindow = 0
+	s := createTestAgentDB(t, reg, opts)
+	app := s.Appender(context.TODO())
+	defer s.Close()
+
+	lbls := labelsForTest(t.Name()+"_histogram", 1)
+	lset := labels.New(lbls[0]...)
+	_, err := app.AppendHistogramCTZeroSample(0, lset, 100, 30, tsdbutil.GenerateTestHistograms(1)[0], nil)
+	require.NoError(t, err)
+
+	_, err = app.AppendHistogramCTZeroSample(0, lset, 100, 100, tsdbutil.GenerateTestHistograms(1)[0], nil)
+	require.Error(t, err)
```

Comment:
I'm curious about why you're appending 2 CTs here. We always append a normal sample after appending CT in the scrape loop. I can't remember a situation where we append 2 CTs in a row 🤔

Reply:
Was testing an error mode, but will try to provide another test case for that.